src/lib/storage/encoded_columns/run_length_column.cpp

```diff
@@ -0,0 +1,103 @@
+#include "run_length_column.hpp"
+
+#include <algorithm>
+
+#include "storage/value_column.hpp"
+#include "type_cast.hpp"
+#include "utils/assert.hpp"
+#include "utils/performance_warning.hpp"
+
+namespace opossum {
+
+template <typename T>
+RunLengthColumn<T>::RunLengthColumn(const std::shared_ptr<const pmr_vector<T>>& values,
+                                    const std::shared_ptr<const pmr_vector<ChunkOffset>>& end_positions,
+                                    const T null_value)
+    : _values{values}, _end_positions{end_positions}, _null_value{null_value} {}
+
+template <typename T>
+std::shared_ptr<const pmr_vector<T>> RunLengthColumn<T>::values() const { return _values; }
+
+template <typename T>
+std::shared_ptr<const pmr_vector<ChunkOffset>> RunLengthColumn<T>::end_positions() const { return _end_positions; }
+
+template <typename T>
+const T RunLengthColumn<T>::null_value() const { return _null_value; }
+
+template <typename T>
+const AllTypeVariant RunLengthColumn<T>::operator[](const ChunkOffset chunk_offset) const {
+  PerformanceWarning("operator[] used");
+
+  const auto end_position_it = std::lower_bound(_end_positions->cbegin(), _end_positions->cend(), chunk_offset);
+  const auto index = std::distance(_end_positions->cbegin(), end_position_it);
+
+  const auto value = (*_values)[index];
+
+  if (value == _null_value) return NULL_VALUE;
+
+  return AllTypeVariant{value};
+}
+
+template <typename T>
+size_t RunLengthColumn<T>::size() const {
+  DebugAssert(_end_positions->size() > 0u, "Column size is never zero.");
+  return _end_positions->back() + 1u;
+}
+
+template <typename T>
+void RunLengthColumn<T>::write_string_representation(std::string& row_string, const ChunkOffset chunk_offset) const {
+  std::stringstream buffer;
+
+  const auto value = (*this)[chunk_offset];
+
+  bool is_null = variant_is_null(value);
+  Assert(!is_null, "This operation does not support NULL values.");
+
+  // buffering value at chunk_offset
+  buffer << value;
+  const uint32_t length = buffer.str().length();
+
+  // writing byte representation of length
+  buffer.write(reinterpret_cast<const char*>(&length), sizeof(length));
+
+  // appending the new string to the already present string
+  row_string += buffer.str();
+}
+
+template <typename T>
+void RunLengthColumn<T>::copy_value_to_value_column(BaseColumn& value_column, ChunkOffset chunk_offset) const {
```

Comment:
What does this method?

Reply:
I think it is used in sort. It copies a value at a given chunk offset to a value column. It’s one of those method that should be removed (#599)

Reply:
Removed!